From c60c5167ca0a1686d4715b83a0b71a6101491c79 Mon Sep 17 00:00:00 2001
From: Ofer Heifetz <oferh@marvell.com>
Date: Mon, 15 May 2017 11:00:02 +0300
Subject: [PATCH 1620/2241] fix: crypto: inside-secure: add unmap on error
 cases

when ring is full or some other mapping fails, mapping must be
removed

Change-Id: Id90ed98a061672413f542ceeddbaae03f3b08d18
Signed-off-by: Ofer Heifetz <oferh@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/39477
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/39529
---
 drivers/crypto/inside-secure/hash.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/crypto/inside-secure/hash.c b/drivers/crypto/inside-secure/hash.c
index a270a29..add6e2f 100644
--- a/drivers/crypto/inside-secure/hash.c
+++ b/drivers/crypto/inside-secure/hash.c
@@ -288,6 +288,11 @@ send_command:
 	return 0;
 
 cdesc_rollback:
+	if (nents)
+		dma_unmap_sg(priv->dev, areq->src,
+			     sg_nents_for_len(areq->src, areq->nbytes),
+			     DMA_TO_DEVICE);
+
 	for (i = 0; i < n_cdesc; i++)
 		safexcel_ring_rollback_wptr(priv, &priv->ring[ring].cdr);
 free_cache:
-- 
2.7.4

